### DISCUSSION TITLE
Some positive feedback & getting other contributors

### DISCUSSION THREAD

**Discussion by florianfelsing** [AllYourBot/hostedgpt#160] [General]
Hey guys,

just got some really positive feedback about our project from my wife. She says that something like this is exactly what she had been looking for, because she's constantly running into the message limit using ChatGPT.

I think it's pretty cool to see end users excited about this project!

As for deploying the project - _if interesting to us at all_ - I could at some point add instructions for deploying HostedGPT using Kamal as well. For now I'll use Render, but this shouldn't be hard to accomplish with Kamal and it would give our users some flexibility in terms of their infrastructure.

**Comment by keithschacht** [comment]
@florianfelsing That’s awesome! Thanks for sharing this. That was one of my original motivations for it as well. The funny thing is that my wife also thinks it’s a really good idea because she only uses ChatGPT maybe 2-4 times per month and doesn’t want to pay $20 for that, but she wants GPT-4 as well. It’s on my list to set her up with her own instance this week so she can stop using my account. :)

So it’s actually both extremes—using it a lot and using it a little—that make it nice to have a different payment model.

**Comment by keithschacht** [comment]
Oh, and related to Kamal, yes! Eventually I wanted to make sure we got Kamal setup and also easy Heroku integration, since enough people still use Heroku. It was actually one of the reasons that Rob went ahead and got docker compose integrated already. We knew we’d eventually want to support docker deploys.

If you want to tackle this because you’re doing it anyway then by all means. If you decide to prioritize it, just comment on this so others know you’re on it:
https://github.com/the-dot-bot/hostedgpt/issues/161

It’s also fine to wait until later, totally up to you. There is quite a long tail of things to improve to get a really solid 1.0. :) And speaking of which, if you have any ideas for finding a couple more people to contribute, I’d love to hear. Keep that in mind.

**Reply by florianfelsing** [reply]
Sure, I'll comment in case I'll be deploying with Kamal already. Otherwise, I agree that it should be fine to wait until later. There's still a lot of stuff to handle.

As for finding more people to contribute, what have you tried already?

What comes to mind immediately:

- Posting in Discord groups
- Our personal networks
- Reaching out to the owners of the top 20 or so results for when you search for "rails open source projects", "ruby open source projects" or related queries - this should work _really_ well in the long-term

**Reply by keithschacht** [reply]
I've reached out to a handful of friends who are rails developers. I have a few people who want to try out the app, but none of them have bitten yet on contributing. Maybe once they start using it they will. :)

I've also tried a couple posts to the r/rails subreddit. [One post last week](https://www.reddit.com/r/rails/comments/1avq2q0/have_you_contributed_to_any_rails_open_source/) explicitly about this project got zero interest. 

I just tried posting something again today with a slightly different strategy: share something I've learned / share interesting parts of the HostedGPT project. This is more "offer value first" and "get them intrigued by the project second". :)

[Feel free to chime in on this post](https://www.reddit.com/r/rails/comments/1b2axyb/im_loving_rails_stimulus_is_this_how_youre_using)

That's a good idea about getting on the pages which come up for "rails open source projects". Likewise, I just tried searching for "open source chatgpt" and there are some good results there. I'll start mining these to see if I can get the project listed on any results.

**Reply by florianfelsing** [reply]
Let me know if I can be of help with the outreach!

The idea with the videos is fantastic and I am pretty sure that it'll work out great over time. They are definitely targeting the right audience. Might record one for each feature I build as well. What I'd suggest is to definitely have 1-2 clear calls to action in each video, as well as in the description below.

Will also reach out to some people in the Rails community that I know and present them the project. One has a fairly popular YouTube channel. Maybe there's a chance for our project to make an appearance there, in whatever form.

Will keep you posted.

**Comment by florianfelsing** [comment]
[google_us_rails-open-source-pr_matching-terms_2024-02-28_20-34-02.csv](https://github.com/the-dot-bot/hostedgpt/files/14439176/google_us_rails-open-source-pr_matching-terms_2024-02-28_20-34-02.csv)

So there is definitely some volume for these terms, check the CSV attached to this comment.

It should not be hard to _be everywhere_ when people search for these. For now I'd probably try to:

- Do the outreach, as mentioned
- Publish a Reddit post with some exact search term (and make it gain some traction)
- Post on Medium and dev.to
- Record a YouTube video targeting some exact search term

Let's coordinate about the outreach - as I said, feel free to delegate some of that to me. The other things, I think we can handle independently.

_Edit: Data source for search volume is Ahrefs._

**Comment by keithschacht** [comment]
I've been thinking about this quite a bit over the past few days and I think I finally have clarity on how to approach this. First, I believe the best way to find people who want to help make this project better is to first get people to use this project. At the the point we have 100 rails engineers all using HostedGPT, people will run into lots of little issues and have new ideas and this is what will create the motivation to fix those issues. My own motivation to help out on other open source projects almost always comes from me using it and running into an issue that I care about. So the most important question is: what will motivate a lot of rails engineers to want to personally use HostedGPT.

This made me realize something important. I've been focused on getting to feature parity between HostedGPT and ChatGPT. This will take many more weeks. But 100% feature match will only create a minimal motivation to use HostedGPT. The primary motivations can really only be: (1) better control over your data and (2) better control over billing/usage.

But I think the thing that will most motivate someone to want to use HostedGPT is that if it has a few compelling features above & beyond ChatGPT. I'll continue to chip away at the Issues in the 0.6 bucket, but I'm going to shift my focus to adding the first big bonus feature: the ability to verbally converse with the bot within your browser & to let it see your screen. I already have a working prototype of this, but it's going to be a chunk of work to get the code cleaned up and integrated into HostedGPT. That's what I'm going to focus on.

I think this will be a pretty wow feature. I've used it myself for a bit and it's really useful. After this is done, then I'll use this "bonus" feature as the best motivation for convincing people to try HostedGPT for themselves.

In the meantime it's worth reaching out to any other rails engineers you know that may be interested in using HostedGPT as-is. But in a couple weeks I think it will become a lot more compelling.

**Reply by florianfelsing** [reply]
_but I'm going to shift my focus to adding the first big bonus feature: the ability to verbally converse with the bot within your browser & to let it see your screen._

This sounds pretty amazing!

And I agree - eventually I see a lot of potential to turn HostedGPT into something that is much more than ChatGPT. A couple of things that come to mind:

- Support LLMs other than the ones provided by OpenAI, all inside the same application, allowing you to pick and choose and not relying on one sole LLM provider
- Creating shared workspaces for teams without having to pay the $25/user per month that OpenAI is charging. Instead, just keep paying for the actual usage and onboard as many folks as you want

**Reply by keithschacht** [reply]
I definitely agree. Those are probably the next biggest ones that I'm excited about for myself, too.

1. On other LLMs. I was pretty disappointed with Gemini Ultra, but I'm hearing good things about Claude 3 already. If it actually starts to get some momentum and people start hearing "this is better than GPT-4" then that could create increasing desire for the ability to switch between models. I sure wish I could just click a button and try it out today without having to sign up for a whole new UI.
2. I haven't thought a lot about teams yet to know what kinds of collaboration features might make sense (e.g. shared conversations). But I've been surprised how many people share their ChatGPT subscription with multiple people. I do, I know others that do, and just today @robacarp was mentioning to me that he shares his with a half dozen people too. There's something about this trend which presents an opportunity. I put this in the bucket of "notice emergent behavior when many users start to use a product in a way that's other than intended." :)
